tuxemon/menu/input.py

```diff
@@ -183,8 +185,9 @@ def add_input_char_and_pop(self, char: str) -> None:
         self.client.pop_state()
 
     def add_input_char(self, char: str) -> None:
-        self.input_string += char
-        self.update_text_area()
+        if len(self.input_string) <= self.char_limit or self.char_limit is None:
```

Comment:
This would fail if char_limit is None because `len(self.input_string) <= self.char_limit` would be evaluated first and would raise a TypeError.  A better way to write this would be to put the `is None` check first.  Python will eval that first and the "short circuit" way conditionals are treated means the latter part `len(self.input_string) <= self.char_limit` would never have have `self.char_limit is None`

```python
if self.char_limit and len(self.input_string) <= self.char_limit:
   ...
```